src/common/routes.js

```diff
@@ -3,21 +3,24 @@ import { Route } from 'react-router';
 import App from './containers/app.js';
 import Builds from './containers/builds.js';
 import BuildDetails from './containers/build-details.js';
-import Dashboard from './containers/dashboard.js';
+import MyRepos from './containers/my-repos.js';
 import Landing from './containers/landing.js';
 import LoginFailed from './containers/login-failed.js';
 import SelectRepositories from './containers/select-repositories.js';
-import EnsureLoggedIn from './containers/ensure-logged-in';
+import EnsureLoggedIn from './containers/routes/ensure-logged-in';
+import EnsureSameUser from './containers/routes/ensure-same-user';
 
 export default (
   <Route component={App}>
     <Route path="/" component={Landing}/>
     <Route component={ EnsureLoggedIn }>
-      <Route path="/dashboard" component={Dashboard}/>
-      <Route path="/dashboard/select-repositories" component={SelectRepositories}/>
+      <Route component={ EnsureSameUser }>
```

Comment:
It would be nice to do it with `onEnter` handler, but it doesn't have access to the store, so wrapping with route and component seems to do the trick.